test/edu-benefits/pages/schoolSelection.unit.spec.jsx

```diff
@@ -1,141 +1,76 @@
-          definitions={formConfig.defaultDefinitions}
-          uiSchema={uiSchemaIfEducationType}/>
-    );
-    const formDOM = findDOMNode(form);
-    const find = formDOM.querySelector.bind(formDOM);
 
-    submitForm(form);
-    expect(Array.from(formDOM.querySelectorAll('.usa-input-error'))).to.have.lengthOf(1);
-    expect(onSubmit.called).to.be.false;
-
-    ReactTestUtils.Simulate.change(find('#root_educationProgram_educationType'), {
-      target: {
-        value: 'college'
+    // Either test for required inputs or not, depending on the required option
+    const requiredInputs = !_.isEmpty(optionsFor[formName].required);
+
+    it(`should have${requiredInputs ? '' : ' no'} required inputs`, () => {
+      const onSubmit = sinon.spy();
+      const form = ReactTestUtils.renderIntoDocument(
+        <DefinitionTester
+            schema={schema}
+            onSubmit={onSubmit}
+            data={{}}
+            definitions={configs[formName].defaultDefinitions}
+            uiSchema={uiSchema}/>
+      );
+      const formDOM = findDOMNode(form);
+      submitForm(form);
+
+      if (requiredInputs) {
+        expect(Array.from(formDOM.querySelectorAll('.usa-input-error'))).to.not.be.empty;
+        expect(onSubmit.called).to.not.be.true;
+      } else {
+        expect(Array.from(formDOM.querySelectorAll('.usa-input-error'))).to.be.empty;
+        expect(onSubmit.called).to.be.true;
       }
     });
 
-    submitForm(form);
-    expect(Array.from(formDOM.querySelectorAll('.usa-input-error'))).to.be.empty;
-    expect(onSubmit.called).to.be.true;
-  });
-});
-
-describe('Edu 5490 schoolSelection', () => {
-  const { schema, uiSchema } = formConfig(fullSchema5490, {
-    fields: [
-      'educationProgram',
-      'educationObjective',
-      'educationStartDate',
-      'restorativeTraining',
-      'vocationalTraining',
-      'trainingState',
-      'educationalCounseling'
-    ],
-    required: ['educationType']
-  });
-
-  it('should render', () => {
-    const form = ReactTestUtils.renderIntoDocument(
-      <DefinitionTester
-          schema={schema}
-          data={{}}
-          definitions={formConfig.defaultDefinitions}
-          uiSchema={uiSchema}/>
-    );
-
-    const formDOM = findDOMNode(form);
-
-    const inputs = Array.from(formDOM.querySelectorAll('input, select, textarea'));
-
-    expect(inputs.length).to.equal(13);
-  });
-
-  it('should have required inputs', () => {
-    const onSubmit = sinon.spy();
-    const form = ReactTestUtils.renderIntoDocument(
-      <DefinitionTester
-          schema={schema}
-          onSubmit={onSubmit}
-          data={{}}
-          definitions={formConfig.defaultDefinitions}
-          uiSchema={uiSchema}/>
-    );
-    const formDOM = findDOMNode(form);
-    submitForm(form);
-
-    expect(Array.from(formDOM.querySelectorAll('.usa-input-error'))).to.not.be.empty;
-    expect(onSubmit.called).to.not.be.true;
+    // TODO: Test filling in the required data
```

Comment:
Hehe, yes. I didn't really want to write the code to fill in _every_ field (if required, of course). Probably should at some point if we keep using this, though.